libcaf_test/caf/test/runner.cpp

```diff
@@ -100,13 +157,15 @@ int runner::run(int argc, char** argv) {
     return std::regex_search(search_string.begin(), search_string.end(),
                              selected);
   };
+  const auto max_runtime = get_or(cfg_, "max-runtime", 0);
   for (auto& [suite_name, suite] : suites_) {
     if (!enabled(*suite_regex, suite_name))
       continue;
     default_reporter->begin_suite(suite_name);
     for (auto [test_name, factory_instance] : suite) {
       if (!enabled(*test_regex, test_name))
         continue;
+      watchdog::start(max_runtime);
```

Comment:
I think there's a conceptual disconnect. Probably the explanation for why you have implemented the watchdog in this way. The watchdog should terminate the program if the whole run reaches the timeout. Your train of thought seemed to be that this is per test. 

Since the watchdog is actually about the program runtime, the initialization should move to *before* the `for` loop and then just be an RAII type.

Also, your implementation is buggy. This will allocate one watchdog for each test that we start, leaking all but the last one.

Reply:
Yes, I was planning to add timer for each test as the task mentioned "aborting test after reaching the time limit". But, as I am calling the watchdog::stop() after completing each test, it should delete existing watchdog without leaking any memory. But, I am moving this implementation to a RAII based approach and running the watchdog once for the whole program.